### PR TITLE
Fix 3745 Trim whitespace on UI entry fields

### PR DESCRIPTION
**Pull request by gally47**
## Description
As described in the issue https://github.com/Graylog2/graylog-plugin-enterprise/issues/3745 not trimming the main fields can create unnecessary user frustration, in this PR the stream title is trimmed with a new helper function **trimObjectFields** that can be used from now on to trim form fields before sending them to the backend.

## Motivation and Context

https://github.com/Graylog2/graylog-plugin-enterprise/issues/3745

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/streams/Stream.jsx [hunk 1]
```diff
@@ -113,6 +114,7 @@ class Stream extends React.Component {
     this.setState({ showEntityShareModal: true });
   };
 
+  // eslint-disable-next-line class-methods-use-this
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/streams/Stream.jsx:117]: we should move the function outside of the Class and that should also fix this linter issue.

File: graylog2-web-interface/src/logic/ObjectHelper.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+// eslint-disable-next-line import/prefer-default-export
+export const trimObjectFields = (obj: object, fieldsToTrim: string[]) => {
```
Comment (ousmaneo) [inline graylog2-web-interface/src/logic/ObjectHelper.ts:18]: I would move this function inside the `ObjectUtils` file we have inside `util` folder.

**Review by ousmaneo — CHANGES_REQUESTED** [review]
Works and fix the issue, I added some small nitpick.

File: graylog2-web-interface/src/logic/ObjectHelper.ts (hunk 2 above)
Comment (gally47) [reply graylog2-web-interface/src/logic/ObjectHelper.ts]: Thx I was searching for smth like that

**Review by ousmaneo — APPROVED** [review]
LGTM! Thank you !

**Pull request merged**